components/StoreLocatorModal.vue

```diff
@@ -0,0 +1,222 @@
+import { useStoreLocations } from "../composables/storeFinder/useStoreLocations"
+import { usePurchaseLocation } from "../composables/storeFinder/usePurchaseLocation"
+
+import { storeLocationGetters } from "@/composables/getters"
+
+export default {
+  name: "LoginModal",
+  components: {
+    SfModal,
+    SfRadio,
+    SfSearchBar,
+    SfAccordion,
+    SfChevron,
+    SfButton,
+  },
+  setup() {
+    const { isStoreLocatorOpen, toggleStoreLocatorModal } = useUiState()
+    const { currentLocation, loadWithNavigator } = useCurrentLocation()
+    const { locations, search: searchStoreLocations } = useStoreLocations()
+    const { set, load: loadPurchaseLocation } = usePurchaseLocation()
+    const selectedStore = ref("")
+
+    useAsync(async () => {
+      await searchStoreLocations()
+    }, null)
+
+    const handleCurrentLocation = async () => {
+      await loadWithNavigator()
+      await searchStoreLocations({
+        latitude: currentLocation.value.latitude,
+        longitude: currentLocation.value.longitude,
+      })
+    }
+
+    const closeModal = () => {
+      toggleStoreLocatorModal()
+      selectedStore.value = ""
+    }
+
+    const storeDetails = computed(() => {
+      return storeLocationGetters.getLocations(locations.value)
+    })
+
+    const handleStoreChange = (locationCode: string) => {
+      selectedStore.value = locationCode
+      set(locationCode)
+      loadPurchaseLocation()
+    }
+
+    return {
+      currentLocation,
+      isStoreLocatorOpen,
+      locations,
+      storeDetails,
+      selectedStore,
+      handleStoreChange,
+      handleCurrentLocation,
+      toggleStoreLocatorModal,
+      closeModal,
+    }
+  },
+}
+</script>
+<style lang="scss" scoped>
+@import "~@storefront-ui/vue/styles";
+.sf-modal {
+  --modal-width: 39.375rem;
+  --modal-content-padding: 0;
+}
+.title {
+  font-weight: 700;
+  font-size: var(--h4-font-size);
+  padding: var(--spacer-2xs) var(--spacer-lg);
+}
+.location-search {
+  display: flex;
+  justify-content: space-between;
+  align-items: stretch;
+  padding: var(--spacer-lg) var(--spacer-base) var(--spacer-2xs) var(--spacer-lg);
+}
+.sf-button__small {
+  margin-left: 2%;
+}
+.Search-bar {
+  flex: 1 !important;
+}
+.current-location {
+  text-decoration: underline;
+  cursor: pointer;
+  font-size: var(--font-size--base);
+  padding: var(--spacer-2xs) var(--spacer-lg);
+}
+.store-count {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-top: 1px solid var(--c-light);
+  padding: var(--spacer-2xs) var(--spacer-sm);
+}
+.store-details {
+  padding: var(--spacer-base) var(--spacer-lg);
+}
+.section-border {
+  border-bottom: 1px solid var(--c-light);
+}
+.sf-accordion-item {
+  &__header {
+    text-decoration: underline;
+  }
+}
+.sf-property__name {
+  text-transform: capitalize;
+}
+
+.sf-radio--transparent {
+  &__content {
+    display: flex !important;
+    justify-content: flex-start;
```

Comment:
.sf-radio--transparent____content is not as per BEM standard, please change it also avoid using !important

Reply:
Done